db/version_edit_handler.cc

```diff
@@ -68,6 +68,30 @@ void VersionEditHandlerBase::Iterate(log::Reader& reader,
   CheckIterationResult(reader, &s);
 
   if (!s.ok()) {
+    if (s.IsCorruption()) {
+      // when we find a Corruption error, something is
+      // wrong with the underlying file. in this case we
+      // want to report the filename, so in here we append
+      // the filename to the Corruption message
+      assert(reader.file());
+
+      // build a new error message
+      std::string message;
```

Comment:
Nit: can we use `ostringstream` to build the message?

Reply:
@riversand963 : if you think it is an improvement, I can offer the following:
```diff
diff --git a/db/version_edit_handler.cc b/db/version_edit_handler.cc
index 3c4acce6a..05b2488ed 100644
--- a/db/version_edit_handler.cc
+++ b/db/version_edit_handler.cc
@@ -10,6 +10,7 @@
 #include "db/version_edit_handler.h"
 
 #include <cinttypes>
+#include <sstream>
 
 #include "db/blob/blob_file_cache.h"
 #include "db/blob/blob_file_reader.h"
@@ -76,21 +77,17 @@ void VersionEditHandlerBase::Iterate(log::Reader& reader,
       assert(reader.file());
 
       // build a new error message
-      std::string message;
+      std::stringstream message;
       // append previous dynamic state message
       const char* state = s.getState();
       if (state != nullptr) {
-        message.append(state);
-      }
-      if (!message.empty() && message.back() != ' ') {
-        // avoid adding redundant whitespace
-        message += " ";
+        message << state;
+        message << ' ';
       }
       // append the filename to the corruption message
-      message += "in file ";
-      message += reader.file()->file_name();
+      message << "in file " << reader.file()->file_name();
       // overwrite the status with the extended status
-      s = Status(s.code(), s.subcode(), s.severity(), message);
+      s = Status(s.code(), s.subcode(), s.severity(), message.str());
     }
     status_ = s;
   }
```
I personally am not convinced that using a stream is a better solution (because of the overhead of stringstream and no access to the string contents during buildup). But I will leave the decision to you and am happy to apply the patch to the PR if you prefer it.
Please let me know. Thanks!

Reply:
Since this is on the failure path which should be rare, I am not very concerned about performance.
In the past, we have been using `snprintf` for cases like this, and we have been advised to use `stringstream` instead.
btw, some online search seems to indicate that `stringstream` avoids building temps and memory allocation, which may not be a problem here as mentioned above. Some newer compilers may also optimize for this case.
I also find stringstream-based version more readable.